Add apis to access primary and gainmap compressed data

ram-mohan opened the pull request:
Also,
- fixed few api names for consistency.
- moved lib version macro to public interface
- documented version semantics inconsistencies

cgohlke commented:
Since `*gain_map*` was renamed to `*gainmap*`, wouldn't it make sense to also rename `UHDR_GAIN_MAP_IMG` to `UHDR_GAINMAP_IMG`?

ram-mohan commented:
> Since `*gain_map*` was renamed to `*gainmap*`, wouldn't it make sense to also rename `UHDR_GAIN_MAP_IMG` to `UHDR_GAINMAP_IMG`?

We renamed the functions from `*gain_map*` to `*gainmap*` because encoder setter functions are using *gainmap*` and decoder setter functions are using *gain_map*`. For consistency we used the same token everywhere. gainmap or gain_map are both fine but use the same everywhere was the idea. So for all functions and variables we have modified to gainmap. enum we left it as is.
